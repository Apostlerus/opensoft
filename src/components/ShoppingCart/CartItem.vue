<template>
  <div class="cart-item">
    <div class="md-layout flex-layout">
      <div class="md-item license-plan-text">
        <md-radio :value="true" v-model="item.active" @change="onSelect">
          <div class="scaled-text">LICENSE PLAN #{{ item.planNumber }}</div>
        </md-radio>
      </div>
      <div class="md-item price-per-license">{{ item.price }} per license
      </div>
    </div>
  </div>
</template>

<script>
export default {
  name: 'cartItem',
  props: {
    item: Object,
  },
  methods: {
    onSelect() {
      this.$emit('onSelect');
    },
  },
};

</script>

<style scoped lang="scss">
  .cart-item {
    .flex-layout {
      display: flex;
      flex-direction: row;
      align-items: center;
      justify-content: space-between;
      padding: 10px 20px;
      &:hover{
        background-color: #E3F2FD;
        outline: 1px solid #90CAF9;
        outline-offset: -1px;
      }
    }
    .price-per-license {
      color: #777;
      &::before {
        font-family: "Proxima Nova";
        content: "\0024";
      }
    }
    .license-plan-text {
      font-weight: 800;
      letter-spacing: 0;
      font-size: 20px;
      line-height: 32px;
      color: #666;
    }
    .scaled-text {
      transform: scale(0.8,1);
      margin: 0 0 0 -18px;
    }
  }
</style>
