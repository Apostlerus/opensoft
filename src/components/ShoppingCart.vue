<template>
  <div class="full-control shopping-cart">
    <CartItem
      v-for="item in items"
      :key="item.id"
      :item="item"
      @onSelect="onSelect(item.id)"
    />
    <md-divider></md-divider>
    <NumberOfLicenses
      :items="licenses"
      v-model="numberOfLicenses"
    />
    <md-divider></md-divider>
    <div class="bottom-group">
      <div
        v-if="getTotalPrice"
        class="total-price"
      >
        <span class="total-price-text">TOTAL: <span class="price">
          ${{ getTotalPrice }}<span class="price-currency">US</span>
        </span></span>
      </div>
      <md-button
        class="md-raised md-primary button-buy"
        :disabled="!getTotalPrice"
      >Buy now</md-button>
      <div
        v-if="getTotalPrice"
        class="selected-plan"
      >
        Selected plan: #{{ getSelectedPlan }}
      </div>
    </div>
  </div>
</template>

<script>
import { mapGetters } from 'vuex';
import CartItem from './ShoppingCart/CartItem.vue';
import NumberOfLicenses from './NumberOfLicenses.vue';

export default {
  name: 'ShoppingCart',
  components: {
    CartItem,
    NumberOfLicenses,
  },
  computed: {
    getTotalPrice() {
      const item = this.items.find(({ active }) => active);
      if (!this.numberOfLicenses || !item) {
        return false;
      }
      return item.price * this.numberOfLicenses;
    },
    getSelectedPlan() {
      const item = this.items.find(({ active }) => active);
      if (!item) {
        return false;
      }
      return item.planNumber;
    },
    ...mapGetters([
      'items',
      'licenses',
    ]),
  },
  mounted() {
    this.$store.dispatch('loadItems');
    this.$store.dispatch('loadNumberOfLicenses');
  },
  data() {
    return {
      numberOfLicenses: 10,
      selectedItem: null,
    };
  },
  methods: {
    onSelect(index) {
      const items = this.items
        .map(item => ({ ...item, active: index === item.id }));
      this.$store.dispatch('updateItems', items);
    },
  },
};
</script>

<style scoped lang="scss">
  .shopping-cart {
    background-color: #f8f8f8;
    width: 590px;
    border: 10px solid #ebebeb;
    padding: 30px 15px;
    .md-button {
      &.button-buy {
        border-radius: 30px;
        width: 150px;
      }
    }
    .bottom-group {
      display: flex;
      align-items: center;
      flex-direction: column;
      .price-currency {
        font-size: small;
        vertical-align: top;
      }
      .price {
        color: #3383A8;
      }
    }
    .selected-plan {
      padding-top: 30px;
      color: #3383A8;
      font-weight: 400;
      font-size: 18px;
    }
    .total-price {
      padding-top: 20px;
      font-weight: 800;
      letter-spacing: 0;
      font-size: 20px;
      line-height: 32px;
      color: #666;
    }
    .total-price-text {
      color: #666;
    }
  }
</style>
