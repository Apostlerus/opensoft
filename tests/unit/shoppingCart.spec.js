import { shallowMount, createLocalVue } from '@vue/test-utils';
import VueMaterial from 'vue-material';
import ShoppingCart from '@/components/ShoppingCart.vue';

describe('ShoppingCart', () => {
  // add this before each
  beforeEach(() => {
    // const localVue = createLocalVue();
    // localVue.use(VueMaterial);
  });
  describe('ShoppingCart.vue', () => {
    const localVue = createLocalVue();
    localVue.use(VueMaterial);
    it('should render content correctly', () => {
      const wrapper = shallowMount(ShoppingCart, {
        localVue,
        mocks: {
          $store: {
            state: {},
            getters: {
              items: [
                {
                  id: 1,
                  planNumber: 1,
                  price: 13,
                  active: false,
                },
                {
                  id: 2,
                  planNumber: 2,
                  price: 22,
                  active: false,
                },
                {
                  id: 3,
                  planNumber: 3,
                  price: 34,
                  active: false,
                },
              ],
              licenses: [1, 2, 3, 10, 100, 1000],
            },
            dispatch: jest.fn(),
          },
        },
      });
      // wrapper.vm.$store.dispatch = jest.fn();
      expect(wrapper.vm.$el).toMatchSnapshot();
    });
  });
  describe('CartItem.vue', () => {

  });
});
