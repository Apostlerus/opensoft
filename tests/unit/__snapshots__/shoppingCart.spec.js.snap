// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShoppingCart ShoppingCart.vue should render content correctly 1`] = `
<div
  class="full-control shopping-cart"
>
  <cartitem-stub
    item="[object Object]"
  />
  <cartitem-stub
    item="[object Object]"
  />
  <cartitem-stub
    item="[object Object]"
  />
   
  <md-divider-stub />
   
  <numberoflicenses-stub
    items="1,2,3,10,100,1000"
    value="10"
  />
   
  <md-divider-stub />
   
  <div
    class="bottom-group"
  >
    <!---->
     
    <md-button-stub
      class="md-raised md-primary button-buy"
      disabled="true"
      mdripple="true"
      type="button"
    >
      Buy now
    </md-button-stub>
     
    <!---->
  </div>
</div>
`;
